src/utils.ts

```diff
@@ -121,19 +135,33 @@ export namespace Utils {
     networkPassphrase: string,
     homeDomains: string | string[],
     webAuthDomain: string,
-  ): { tx: Transaction; clientAccountID: string; matchedHomeDomain: string } {
+  ): {
+    tx: Transaction;
+    clientAccountID: string;
+    matchedHomeDomain: string;
+    memo: string | null;
+  } {
     if (serverAccountID.startsWith("M")) {
       throw Error(
         "Invalid serverAccountID: multiplexed accounts are not supported.",
       );
     }
 
-    const transaction = TransactionBuilder.fromXDR(
-      challengeTx,
-      networkPassphrase,
-    );
-
-    if (!(transaction instanceof Transaction)) {
+    let transaction;
+    try {
+      transaction = new Transaction(challengeTx, networkPassphrase, true);
+    } catch {
+      try {
+        transaction = new FeeBumpTransaction(
+          challengeTx,
+          networkPassphrase,
+          true,
+        );
```

Comment:
❓ Why do we need to move away from `TransactionBuilder.fromXDR`?

Reply:
Also, challenge txs should never be fee bump transactions, so if we do need to move away it is unnecessary to parse as a fee bump transaction here, and in fact preferable if we don't. I'd consider it a bug, albeit a meaningless one I think, that this function accepts a fee bump.

Reply:
>❓ Why do we need to move away from TransactionBuilder.fromXDR?

The builder class doesn't support muxed accounts, see https://github.com/stellar/js-stellar-base/pull/470

> I'd consider it a bug, albeit a meaningless one I think, that this function accepts a fee bump.

I was just preserving existing behavior, which included returning a fee-bump related error message if the `!(transaction instanceof Transaction)` returned true. I can remove this message and simply return a message indicating that the transaction wasn't deserialize-able as a `Transaction` object.

Reply:
> The builder class doesn't support muxed accounts, see [stellar/js-stellar-base#470](https://github.com/stellar/js-stellar-base/pull/470)

Is waiting to merge this PR till that PR is merged an option? 

Solving the fee bump issue is probably not worth it. It looks like other SDKs also allow the tx to be a fee bump, although I don't think that has any meaningful impact.

Reply:
> I don't think keeping the use of the `TransactionBuilder` class provides any advantages over using the `Transaction` class.

👍🏻

Reply:
Except the fact that the docs for the latter say not to use it, heh:

> Use TransactionBuilder to build a transaction object. If you have an object or base64-encoded string of the transaction envelope XDR, use TransactionBuilder.fromXDR.

Reply:
One thing I note is different is `TransactionBuilder.fromXDR` also does base64 decoding, but it doesn't look like that is happening here. 🤔

Reply:
If there's urgency to merge this before the bump to js-base, can you file an issue so this doesn't get forgotten? I'd prefer to have it use the builder

Reply:
I'll do that now 👍

Reply:
https://github.com/stellar/js-stellar-base/issues/471